src/serac/physics/solid_mechanics.hpp

```diff
@@ -1004,12 +1094,37 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
 
     auto& lin_solver = nonlin_solver_->linearSolver();
 
-    lin_solver.SetOperator(*J_);
+    if (!contact_.haveLagrangeMultipliers()) {
+      lin_solver.SetOperator(*jacobian);
+
+      lin_solver.Mult(dr_, du_);
+      displacement_ += du_;
+
+      nonlin_solver_->solve(displacement_);
+    } else {
+      lin_solver.SetOperator(*J_contact_);
+
+      int               disp_size = displacement_.Size();
+      mfem::Array<int>  block_offsets({0, disp_size, disp_size + contact_.numPressureTrueDofs()});
```

Comment:
A comment block showing the linearized system of equations that you are solving would be good.